components/brave_wallet/browser/asset_discovery_manager.h

```diff
@@ -109,20 +107,32 @@ class AssetDiscoveryManager : public mojom::KeyringServiceObserver {
       const base::flat_set<std::string>& discovered_contract_addresses,
       std::vector<mojom::BlockchainTokenPtr> sol_token_registry);
 
-  void DiscoverEthAssets(const std::string& chain_id,
-                         mojom::CoinType coin,
-                         const std::vector<std::string>& account_addresses,
-                         bool triggered_by_accounts_added,
-                         const std::string& from_block,
-                         const std::string& to_block);
-
-  void OnGetEthTokenRegistry(const std::string& chain_id,
-                             const std::vector<std::string>& account_addresses,
-                             std::vector<mojom::BlockchainTokenPtr> user_assets,
-                             bool triggered_by_accounts_added,
-                             const std::string& from_block,
-                             const std::string& to_block,
-                             std::vector<mojom::BlockchainTokenPtr> token_list);
+  void DiscoverEthAssets(const std::vector<std::string>& account_addresses,
+                         bool triggered_by_accounts_added);
+
+  void OnGetEthTokenRegistry(
+      const std::vector<std::string>& account_addresses,
+      const std::vector<mojom::BlockchainTokenPtr>& user_assets,
+      bool triggered_by_accounts_added,
+      TokenListMap token_list_map);
+
+  void OnGetERC20TokenBalances(
+      base::OnceCallback<void(std::map<std::string, std::vector<std::string>>)>
+          barrier_callback,
+      const std::string& chain_id,
+      bool triggered_by_accounts_added,
+      const std::vector<std::string>& contract_addresses,
+      const std::vector<absl::optional<std::string>>& balances,
+      mojom::ProviderError error,
+      const std::string& error_message);
+
+  void MergeDiscoveredEthAssets(
+      base::flat_map<std::string,
+                     base::flat_map<std::string, mojom::BlockchainTokenPtr>>
```

Comment:
Yes it's the same type definition, but since the keys are not the same (chainID, vs coin.chainID), I thought it might be misleading to use the same type.

Reply:
I don't think the types match in this case.  From blockchain_list_parser.h, this is a TokenListMap
```
using TokenListMap =
    base::flat_map<std::string, std::vector<mojom::BlockchainTokenPtr>>;
```

But in this case, the first parameter of MergeDiscoveredAssets is a map of strings to a map of strings to BlockchainTokenPtrs (no vector).  The main purpose of this paramter is to be a look up for a BlockchainTokenPtr for a given chain ID and contract address.